### PR TITLE
DT-383 convert auth user service to kotlin

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/auth/model/Group.java [hunk 1]
```diff
@@ -38,4 +38,20 @@ public Group(final String groupCode, final String groupName) {
         this.groupCode = groupCode;
         this.groupName = groupName;
     }
+
+    public UUID getId() {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/auth/model/Group.java]: lombok / kotlin interoperability

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/UserDetailsController.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ open class UserDetailsController(private val authUserService: AuthUserService,
 
     return try {
       val username = getUserName(authentication)
-      authUserService.amendUser(username, firstName, lastName)
+      authUserService.amendUser(username, firstName!!, lastName!!)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/UserDetailsController.kt]: will fix in subsequent PR - blank validation needs moving to service

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserController.java [hunk 3]
```diff
@@ -246,10 +245,6 @@ private String createInitialPasswordUrl(@ApiIgnore final HttpServletRequest requ
             return ResponseEntity.noContent().build();
         } catch (final EntityNotFoundException e) {
             return ResponseEntity.notFound().build();
-        } catch (final AmendUserException e) {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserController.java]: not ever thrown by service

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+  @Throws(AuthUserGroupRelationshipException::class)
+  private fun changeUserEnabled(username: String, enabled: Boolean, admin: String, authorities: Collection<GrantedAuthority>) {
+    val user = userRepository.findByUsernameAndMasterIsTrue(username)
+        .orElseThrow { EntityNotFoundException("User not found with username $username") }
+    maintainUserCheck.ensureUserLoggedInUserRelationship(admin, authorities, user)
+    user.isEnabled = enabled
+    // give user 7 days grace if last logged in more than x days ago
+    if (user.lastLoggedIn.isBefore(LocalDateTime.now().minusDays(loginDaysTrigger.toLong()))) {
+      user.lastLoggedIn = LocalDateTime.now().minusDays(loginDaysTrigger - 7.toLong())
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt]: could you have written `LocalDateTime.now().minusDays(loginDaysTrigger - 7l)`  ?

Comment (petergphillips) [reply src/main/java/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt]: yes, good spot.  I just relied on the java -> kotlin conversion here.  Will fix in next PR

**Pull request merged**